server/src/internalClusterTest/java/org/elasticsearch/discovery/DiscoveryDisruptionIT.java

```diff
@@ -235,6 +229,12 @@ public void testJoinWaitsForClusterApplier() throws Exception {
         }, ActionListener.wrap(() -> {}));
         barrier.await(10, TimeUnit.SECONDS);
 
+        // drop the victim from the cluster with a network disruption
```

Comment:
Hi @DaveCTurner , LGTM. I understand the issue that the victim might have sent the join request before making its network fail. However, I am still not sure why that was the reason that the second `barrier.await(10, TimeUnit.SECONDS);` was the one timing out in the original issue's build scan ( https://gradle-enterprise.elastic.co/s/22kfr6xkperam/tests/:server:internalClusterTest/org.elasticsearch.discovery.DiscoveryDisruptionIT/testJoinWaitsForClusterApplier ) ? Even if the join request was sent earlier, shouldn't the barriers be passing either way? I would have expected a failure at `ensureStableCluster(2, masterName);` since the cluster would have 3 nodes instead of 2.

Reply:
`ensureStableCluster(2, masterName)` waits for up to 30s for the cluster to have 2 nodes (by calling the cluster health API with a 30s timeout) and the barrier wait was timing out in this time. But then even if it didn't time out, `ensureStableCluster` throws an `AssertionError` eventually so we'd never get to the second `barrier.await` on the main test thread.

I think you should be able to reproduce this with the changes in [this comment](https://github.com/elastic/elasticsearch/issues/86974#issuecomment-1160314795).

Reply:
Ah I see thank you!